pkg/restore/import.go

```diff
@@ -159,10 +180,22 @@ func (importer *FileImporter) Import(file *backup.File, rewriteRules *restore_ut
 		if err != nil {
 			return errors.Trace(err)
 		}
+		stores := make(map[uint64]struct{})
 		// Try to download and ingest the file in every region
 		for _, regionInfo := range regionInfos {
 			var downloadMeta *import_sstpb.SSTMeta
 			info := regionInfo
+			// set store speed limit
+			for _, peer := range info.Region.Peers {
+				storeID := peer.StoreId
+				if _, ok := stores[storeID]; !ok {
+					err := importer.setDownloadSpeedLimit(storeID, importer.rateLimit)
```

Comment:
Well this looks too complicated.

The current implementation will call `SetDownloadSpeedLimit` for the associated store *every time* a backup file is processed. So if there are N files this API would be called N times. While the expected design is just call this once per store at start up.

The current code can handle a new store appearing while restoring... but I'm not sure if such scenario is worth supporting.

Reply:
yes, it looks a little complicated. because I have no idea how to get stores before impoting files.I have moved `stores` outside to avoid call N times.

Reply:
@lhy1024 

> because I have no idea how to get stores before impoting files.

You call `GetAllStores()` from a `pd.Client`.

Reply:
thanks